core/trino-main/src/main/java/io/trino/operator/exchange/LocalExchangeSource.java

```diff
@@ -41,8 +42,11 @@
 
     private final Consumer<LocalExchangeSource> onFinish;
 
-    private final BlockingQueue<PageReference> buffer = new LinkedBlockingDeque<>();
```

Comment:
so we never relied on buffer to be blocking?

Reply:
No. `new LinkedBlockingDeque<>()` is unbounded